source/extensions/common/dynamic_forward_proxy/dns_cache_manager_impl.cc

```diff
@@ -31,6 +31,15 @@ DnsCacheSharedPtr DnsCacheManagerImpl::getCache(
   return new_cache;
 }
 
+DnsCacheSharedPtr DnsCacheManagerImpl::lookUpCacheByName(absl::string_view cache_name) {
```

Comment:
Sorry forgot this from before, can you ASSERT this is thread safe / called from the main thread? I think there are other relevant assertions in this file.